rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -151,64 +164,70 @@ class ReplayRSpace[F[_]: Sync, C, P, A, K](
               }
       } yield r
 
-    def getCommOrDataCandidates(comms: Seq[COMM]): F[Either[COMM, Seq[DataCandidate[C, A]]]] = {
-      type COMMOrData = Either[COMM, Seq[DataCandidate[C, A]]]
-      def go(comms: Seq[COMM]): F[Either[Seq[COMM], Either[COMM, Seq[DataCandidate[C, A]]]]] =
+    def getCommAndDataCandidates(
+        comms: Seq[COMM]
+    ): F[Option[(COMM, Seq[DataCandidate[C, A]])]] = {
+      type COMMOrData = Either[COMM, (COMM, Seq[DataCandidate[C, A]])]
+      def go(
+          comms: Seq[COMM]
+      ): F[Either[Seq[COMM], Either[COMM, (COMM, Seq[DataCandidate[C, A]])]]] =
         comms match {
           case Nil =>
             val msg = "List comms must not be empty"
             logger.error(msg)
             Sync[F].raiseError(new IllegalArgumentException(msg))
           case commRef :: Nil =>
             runMatcher(commRef).map {
-              case Some(x) => x.asRight[COMM].asRight[Seq[COMM]]
-              case None    => commRef.asLeft[Seq[DataCandidate[C, A]]].asRight[Seq[COMM]]
+              case Some(x) => (commRef, x).asRight[COMM].asRight[Seq[COMM]]
```

Comment:
request: could you name `x` properly? I know it's not your change, but it makes my OCD itch.

Reply:
Can do